CMakePresets.json

```diff
@@ -0,0 +1,173 @@
+{
+  "version": 3,
+  "configurePresets": [
+    {
+      "name": "base",
+      "displayName": "base preset",
+      "generator": "Ninja",
+      "binaryDir": "${sourceDir}/build/${presetName}",
```

Comment:
Having multiple build dirs isn't bad, but surprising compared to previous behavior. Maybe we should drop the `${presetName}` here and in `LOG_DIR`?

Reply:
Kinda agree, as this change would break other scripts that assume everything lives directly under `build/` (like `make install`?) It would also make it harder mixing presets and just `make`.

On the other hand, having separate build directories can be useful when testing PRs so you don't clobber your main build directory.

In sum, this merits a dedicated discussion and PR.